rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/CovidRepeatingGroupFactoryTest.java

```diff
@@ -0,0 +1,35 @@
+package io.ona.rdt.widget;
+
+import android.view.View;
+
+import com.rengwuxian.materialedittext.MaterialEditText;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+import io.ona.rdt.R;
+import io.ona.rdt.robolectric.widget.WidgetFactoryRobolectricTest;
+
+public class CovidRepeatingGroupFactoryTest extends WidgetFactoryRobolectricTest {
+
+    @Test
+    public void verifyEditTextListeners() throws Exception {
+        CovidRepeatingGroupFactory covidRepeatingGroupFactory = new CovidRepeatingGroupFactory();
+        JSONObject jsonObject = new JSONObject("{\"value\": [], \"key\": \"test\", \"reference_edit_text_hint\": \"\"}");
+        List<View> views = covidRepeatingGroupFactory.getViewsFromJson("step1", jsonFormActivity, new JsonFormFragment(), jsonObject, null);
+        MaterialEditText materialEditText = views.get(0).findViewById(R.id.reference_edit_text);
+        materialEditText.getOnFocusChangeListener().onFocusChange(materialEditText, false);
+        Assert.assertEquals(1, views.size());
```

Comment:
Could we verify that  `addOnDoneAction` is called?

Reply:
yes, but in that scenario, we have to use a real fragment with presenter and interactor.

Reply:
ex: FragmentScenario not working i have checked

Reply:
Why do we need a real fragment?

Reply:
because when we trigger the method `addOnDoneAction` then this initiates an async task 'AttachRepeatingGroupTask` and this task builds a repeating group widget with the help of presenter and interactor.

Reply:
you can spy the factory if you don't want the method executed but still need to verify it's called.

Reply:
verified `addOnDoneAction` called